contracts/bridge/src/bridge.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import {ERC20} from "lib/openzeppelin-contracts/contracts/token/ERC20/ERC20.sol";
+
+contract Bridge is ERC20 {
```

Comment:
Instead of making the bridge an ERC20, it is better to separate the two contract deployments where the bridge contract calls into the erc20 contract to mint new tokens and for users to be able to burn their tokens. I think this is achievable with https://docs.openzeppelin.com/contracts/4.x/api/token/erc20#ERC20PresetMinterPauser

Reply:
I looked into this today, and it took me some time to understand why it wasn't working for me.. So the problem is that if you deploy the ERC20 contract separately, the bridge contract can not transfer or move user's funds, unless prior to calling the bridge contract, they explicitly call `erc20.approve` to allow the bridge to take control of their funds. Idk if this is a common pattern in the ethereum world, but it seems less user friendly to me

Reply:
Yep, it's common to issue a generic approve in the ETH world so that is fine.

Better use ERC20 due to some wallet incompatibility and also when others integrate with the bridge, better for them to use the ERC20 interface.